tool/plugin/lib/plugin.dart

```diff
@@ -0,0 +1,535 @@
+class AntBuildCommand extends ProductCommand {
+  final BuildCommandRunner runner;
+
+  AntBuildCommand(this.runner);
+
+  String get name => 'build';
+
+  String get description => 'Build a deployable version of the Flutter plugin, '
+      'compiled against the specified artifacts.';
+
+  Future<int> run() async {
+    super.run();
+    var value;
+    for (var spec in specs) {
+      await spec.artifacts.provision(); // Not needed for ant script.
+      await deleteBuildContents();
+      value = await ant(spec);
+      if (value != 0) {
+        return value;
+      }
+      await moveToArtifacts(spec);
+    }
+    return value;
+  }
+}
+
+/// Build deployable plugin files. If the --release argument is given
+/// then perform additional checks to verify that the release environment
+/// is in good order.
+class BuildCommand extends ProductCommand {
+  final BuildCommandRunner runner;
+
+  BuildCommand(this.runner);
+
+  String get name => 'build';
+
+  String get description => 'Build a deployable version of the Flutter plugin, '
+      'compiled against the specified artifacts.';
+
+  Future<int> run() async {
+    super.run();
+    for (var spec in specs) {
+      await spec.artifacts.provision();
+
+      separator('Building flutter-intellij.jar');
+
+      List<File> jars = []..addAll(
+          findJars('${spec.dartPlugin.outPath}/lib'))..addAll(
+          findJars('${spec.product.outPath}/lib')); // TODO: also, plugins
+
+      List<String> sourcepath = [
+        'src',
+        'resources',
+        'gen',
+        'third_party/intellij-plugins-dart/src'
+      ];
+      createDir('build/classes');
+
+      await runner.javac(
+        sources: sourcepath.expand(findJavaFiles).toList(),
+        sourcepath: sourcepath,
+        destdir: 'build/classes',
+        classpath: jars.map((j) => j.path).toList(),
+      );
+
+      // copy resources
+      copyResources(from: 'src', to: 'build/classes');
+      copyResources(from: 'resources', to: 'build/classes');
+      copyResources(from: 'gen', to: 'build/classes');
+      copyResources(
+          from: 'third_party/intellij-plugins-dart/src', to: 'build/classes');
+
+      // create the jar
+      await jar('build/classes', 'build/flutter-intellij.jar');
+    }
+    return 0;
+  }
+}
+
+/// Build the tests if necessary then
+/// run them and return any failure code.
+class TestCommand extends ProductCommand {
+  final BuildCommandRunner runner;
+
+  TestCommand(this.runner) {
+    argParser.addFlag(
+        'unit', abbr: 'u', defaultsTo: true, help: 'Run unit tests');
+    argParser.addFlag(
+        'integration', abbr: 'i',
+        defaultsTo: false,
+        help: 'Run integration tests');
+  }
+
+  String get name => 'test';
+
+  String get description => 'Run the tests for the Flutter plugin.';
+
+  Future<int> run() async {
+    super.run();
+    for (var spec in specs) {
+      await spec.artifacts.provision();
+
+      // TODO(devoncarew): implement
+    }
+    return 0;
```

Comment:
perhaps currently fail with `'unimplemented'`?

Reply:
I can change that for all these.